src/braket/tracking/tracker.py

```diff
@@ -0,0 +1,280 @@
+                also includes the duration of the simulation.
+
+        Example:
+            >>> tracker.quantum_tasks_statistics()
+            {'qpu_device_foo':
+                {'shots' : 1000,
+                 'tasks' : { 'COMPLETED' : 4,
+                             'QUEUED' : 1 },
+                },
+             'simulator_device_bar':
+                {'shots' : 1000
+                 'tasks' : { 'COMPLETED' : 2,
+                              'CREATED' : 1},
+                 'execution_duration' : datetime.timedelta(seconds=5, microseconds=654321),
+                 'billed_execution_duration' : datetime.timedelta(seconds=6, microseconds=123456)}}
+        """
+        stats = {}
+        for _, details in self._resources.items():
+
+            device_stats = stats.get(details["device"], {})
+
+            shots = device_stats.get("shots", 0) + details["shots"]
+            device_stats["shots"] = shots
+
+            task_states = device_stats.get("tasks", {})
+            task_states[details["status"]] = task_states.get(details["status"], 0) + 1
+            device_stats["tasks"] = task_states
+
+            if "execution_duration" in details:
+                duration = (
+                    device_stats.get("execution_duration", timedelta(0))
+                    + details["execution_duration"]
+                )
+                billed_duration = (
+                    device_stats.get("billed_execution_duration", timedelta(0))
+                    + details["billed_duration"]
+                )
+
+                device_stats["execution_duration"] = duration
+                device_stats["billed_execution_duration"] = billed_duration
+
+            stats.setdefault(details["device"], {}).update(device_stats)
+
+        return stats
+
+
+def _get_qpu_task_cost(task_arn: str, details: dict) -> Decimal:
+    if details["status"] in ["FAILED", "CANCELLED"]:
+        return Decimal(0)
+    task_region = task_arn.split(":")[3]
+
+    search_dict = {"Region Code": task_region}
+
+    device_name = details["device"].split("/")[-1]
+    device_name = device_name[0].upper() + device_name[1:]
+    if "2000Q" in device_name:
+        device_name = "2000Q"
+    elif "Advantage_system" in device_name:
+        device_name = "Advantage_system"
+
+    if details["job_task"]:
+        search_dict["Device Name"] = device_name
+        shot_product_family = "Braket Managed Jobs QPU Task Shot"
+        task_product_family = "Braket Managed Jobs QPU Task"
+    else:
+        search_dict["DeviceName"] = device_name  # The difference in spelling is intentional
+        shot_product_family = "Quantum Task-Shot"
+        task_product_family = "Quantum Task"
+
+    search_dict["Product Family"] = shot_product_family
+    shot_prices = price_search(**search_dict)
+    if len(shot_prices) != 1:
+        raise ValueError(f"Found {len(shot_prices)} products matching {search_dict}")
```

Comment:
Do we have any cases where `len(shot_prices) > 1` ?

Reply:
We should not. The search_dict should be constructed to find a unique entry from the products